### ISSUE TITLE
Rerun repeated uniqueN test

### ISSUE THREAD

**Issue by mattdowle** [Rdatatable/data.table#3438]
Was reported here : https://github.com/Rdatatable/data.table/issues/3395#issuecomment-464436815
I said I'd follow up here: https://github.com/Rdatatable/data.table/pull/3435#issuecomment-468415545
Double check all those results under the new default.

**Comment by jangorecki** [comment]
Up to date timings below. There is still a problem in uniqueN by group.
We basically need what @mattdowle described in https://github.com/Rdatatable/data.table/pull/3743#issuecomment-525023671 and #1120 would also be nice.
```r
library(data.table)
N_X = 1e6
n_day = 60
n_clientid = 1e5
n_Platform = 7
X = data.table(
  day = sample(1:n_day, N_X, TRUE),
  clientid = as.character(sample(1:n_clientid, N_X, TRUE)),
  Platform = as.character(sample(1:n_Platform, N_X, TRUE))
)

setDTthreads(NULL) # default
getDTthreads()
#[1] 10
system.time(
  X[, .(x = uniqueN(day) - 1L,
        first_active_day = min(day),
        last_active_day = max(day)),
    by = .(Platform, clientid)]
)
#   user  system elapsed 
#172.084   1.640  18.179 

setDTthreads(1)
getDTthreads()
#[1] 1
system.time(
  X[, .(x = uniqueN(day) - 1L,
        first_active_day = min(day),
        last_active_day = max(day)),
    by = .(Platform, clientid)]
)
#   user  system elapsed 
# 10.608   0.596  11.206 

setDTthreads(0)
getDTthreads()
#[1] 20
system.time(
  X[, .(x = uniqueN(day) - 1L,
        first_active_day = min(day),
        last_active_day = max(day)),
    by = .(Platform, clientid)]
)
#still computing
```

**Comment by mattdowle** [comment]
Closed by #4484. See benchmark near the top of the top comment here: https://github.com/Rdatatable/data.table/pull/4484#issue-422469250
